lib/puppet/provider/service/systemd.rb

```diff
@@ -50,37 +50,21 @@ def get_start_link_count
   end
 
   def enabled?
-    begin
-      systemctl_info = systemctl(
-         'show',
-         @resource[:name],
-         '--property', 'LoadState',
-         '--property', 'UnitFileState',
-         '--no-pager'
-      )
-
-      svc_info = Hash.new
-      systemctl_info.split.each do |svc|
-        entry_pair = svc.split('=')
-        svc_info[entry_pair.first.to_sym] = entry_pair.last
-      end
-
-      # The masked state is equivalent to the disabled state in terms of
-      # comparison so we only care to check if it is masked if we want to keep
-      # it masked.
-      #
-      # We only return :mask if we're trying to mask the service. This prevents
-      # flapping when simply trying to disable a masked service.
-      return :mask if (@resource[:enable] == :mask) && (svc_info[:LoadState] == 'masked')
-      return :true if svc_info[:UnitFileState] == 'enabled'
-      if Facter.value(:osfamily).downcase == 'debian'
-        ret = debian_enabled?(svc_info)
-        return ret if ret
-      end
-    rescue Puppet::ExecutionFailure
-      # The execution of the systemd command can fail for quite a few reasons.
-      # In all of these cases, the failure of the query indicates that the
-      # service is disabled and therefore we simply return :false.
+    cmd = [command(:systemctl), 'is-enabled', @resource[:name]]
+    output = execute(cmd, :failonfail => false).strip
+
+    # The masked state is equivalent to the disabled state in terms of
+    # comparison so we only care to check if it is masked if we want to keep
+    # it masked.
+    #
+    # We only return :mask if we're trying to mask the service. This prevents
+    # flapping when simply trying to disable a masked service.
+    return :mask if (@resource[:enable] == :mask) && (output == 'masked')
+    return :true if output == 'enabled'
+    return :false if ['disabled', 'linked', 'static', 'indirect', 'masked'].include? output
+    if (output.empty?) && (Facter.value(:osfamily).downcase == 'debian')
```

Comment:
@MikaelSmith `systemctl is-enabled' should return `enabled-runtime`, `linked-runtime`, or `masked-runtime` per the [https://www.freedesktop.org/software/systemd/man/systemctl.html#is-enabled%20NAME...](systemctl man page).

This was something I noticed when developing this patch. I didn't worry about it because as you noted the previous implementation didn't. I forgot to mention this when I submitted the PR.